File: src/com/leetcode_practice/problems/medium/RotateList_61.java
package com.leetcode_practice.problems.medium;

public class RotateList_61 {
    public ListNode rotateRight(ListNode head, int k) {
        if (head == null || head.next == null || k == 0)
            return head;
        int len = 1;
        ListNode tail = head;
        while (tail.next != null) {
            len++;
            tail = tail.next;
        }
        k = k % len;
        if (k == 0)
            return head;

        // Find new tail (len - k-1)
        ListNode newTail = head;
        for (int i = 0; i < len - k - 1; i++) {
            newTail = newTail.next;
        }

        ListNode newHead = newTail.next;
        newTail.next = null;
        tail.next = head;
        return newHead;

    }
}

File: src/com/leetcode_practice_py/medium/RotateList_61.py
# Definition for singly-linked list.
from typing import Optional


class ListNode:
     def __init__(self, val=0, next=None):
        self.val = val
        self.next = next
        
class RotateList_61:
    def rotateRight(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        if(head is None or  head.next is None or k == 0):
            return head
        len = 1
        tail = head
        while(tail.next is not None):
            len+=1
            tail = tail.next
        
        k = k % len
        if(k == 0):
            return head
        newTail = head
        for i in range(len-k-1):
            newTail = newTail.next
        
        newHead = newTail.next
        newTail.next = None
        tail.next = head
        return newHead